feat: Spring Security

ponomarevmv8 opened the pull request:
Добавление авторизации и
аунтефикации.
Добавление сервера регистрации
и аунтефикации пользователей

PonomarevMV committed:
feat: Spring Security

Добавление авторизации и
аунтефикации.
Добавление сервера регистрации
и аунтефикации пользователей

sorokinuniversity commented on src/main/java/ponomarev/dev/eventmanager/security/SecurityConfiguration.java:
```diff
@@ -0,0 +1,94 @@
+    @Autowired
+    private JwtTokenFilter jwtTokenFilter;
+
+
```
здесь надо добавить 
@ Bean
    public WebSecurityCustomizer webSecurityCustomizer() {
        return web -> web.debug(true).ignoring()
                .requestMatchers("/css/**",
                        "/js/**",
                        "/img/**",
                        "/lib/**",
                        "/favicon.ico",
                        "/swagger-ui/**",
                        "/v2/api-docs",
                        "/v3/api-docs",
                        "/configuration/ui",
                        "/swagger-resources/**",
                        "/configuration/security",
                        "/swagger-ui.html",
                        "/webjars/**",
                        "/v3/api-docs/swagger-config",
                        "/openapi.yaml"
                );
    }
чтобы мы могли заходить в сваггер без аутентификации

sorokinuniversity commented on src/main/java/ponomarev/dev/eventmanager/security/jwt/JwtAuthenticationService.java:
```diff
@@ -0,0 +1,37 @@
+        authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(
+                userCredentials.login(),
+                userCredentials.password()
+        ));
```
здесь неплохо бы сразу проверять есть ли такой юзер в базе, если нет -> бросать ошибку

sorokinuniversity commented on src/main/java/ponomarev/dev/eventmanager/security/CustomAccessDeniedHandler.java:
```diff
@@ -0,0 +1,48 @@
+        var stringMessage = mapper.writeValueAsString(message);
+
+        response.setStatus(HttpStatus.FORBIDDEN.value());
+        response.setContentType(MediaType.APPLICATION_JSON_VALUE);
```
может проблема в моей системе, но вместо сообщения в сваггере вижу знаки вопроса
response.setCharacterEncoding("UTF-8"); - помогло. Проверь везде, есть ли у тебя такая проблема - если да то надо пройтись и скорректировать кодировку

sorokinuniversity commented on src/main/java/ponomarev/dev/eventmanager/user/api/UserRegistration.java:
```diff
@@ -0,0 +1,19 @@
+        String login,
+
+        @NotBlank
+        @Min(4)
```
min это для чисел.  для стринги работает некорректно и не дает зарегаться с паролем из букв

sorokinuniversity reviewed (commented):
Все хорошо, поправь и можно идти дальше

PonomarevMV committed:
feat: Spring Security

Добавление авторизации и
аунтефикации.
Добавление сервера регистрации
и аунтефикации пользователей
